### ISSUE TITLE
Output file silently ignores the delimiter passed

### ISSUE THREAD

**Issue by Minipada** [fluent/fluent-bit#7284]
## Bug Report

**Describe the bug**
When using output file, it silently ignores the delimiter parameter.

Discussion started on Slack: https://fluent-all.slack.com/archives/C0CTQGHKJ/p1682626858152819

**To Reproduce**
- Steps to reproduce the problem:
Command:
```bash
./fluent-bit -i dummy -o file -p format=csv -p delimiter=';' -p file=my_file
```

```bash
cat my_file
1682626714.581559350,"dummy"
1682626715.581501656,"dummy"
```

It still uses the comma

**Expected behavior**
Either use the semicolon (or any other character passed, but semicolon should at least work for csv, no?) or throw an error

**Your Environment**
* Version used: 2.0.7
* Configuration: None
* Operating System and version: Ubuntu 22
* Filters and plugins: None

**Comment by leonardo-albertovich** [comment]
You are right, for some reason the output plugin only accepts a limited set of options as delimiters and silently discards anything else https://github.com/fluent/fluent-bit/blob/master/plugins/out_file/file.c#L60

I think the right procedure here would be checking the commit logs for file.c to determine if there is a strong reason for this, and if there is not then make it so it retains the current behavior (interpreting certain strings such as "\t", "tab", "comma" or "space" replacing them with the character) and if the user provided value falls out of those use it literally.

The one thing you need to keep in mind is that this function is returning a constant string which is not released by the plugins destructor and with the change you want to make you would have to return a new dynamically allocated string and add the appropriate code to release it upon destruction.

If you want to make the process faster and more likely to happen I'd encourage you to try to make the PR yourself.

I think in this case you would have to make the PR against master (2.1) and then there's a fair chance it could be backported since it's more of a fix than a feature in my opinion.

If you decide to give it a try and need guidance feel free to reach out to me in the fluent slack and I'll assist you through the process.

Thanks for reporting the issue!
